### PR TITLE
Modify the path include guard to fix some false-positives

### LINKED ISSUE #3045: [BUG] Regex for checking for relative includes also checks comments
**Describe the bug**
Basically, the check for relative includes using some regex applies to the whole line, and not just the content of the angular brackets in the include. Therefore, if you have a comment on the same line containing a set of angular brackets with at least 2 periods in it, it will misinterpret it as being the actual include path and emit `<stdin>:1:1: no absolute or relative includes please`.

**To Reproduce**
MRE:
```cpp
#include <cstdio>  // <..>
int main() {}
```
The first line will result in `<stdin>:1:1: no absolute or relative includes please`

**Expected behavior**
Generally it's not really an issue (especially if you know why it happens), but it prevents code like the following:
```cpp
#include <cassert>     // for assert(...)
#include <utility>     // for std::exchange(...)
#include <stdexcept>   // for std::out_of_range
#include <ranges>      // for std::ranges::range<...> and std::ranges::range_type_v<...>
#include <type_traits> // for std::is_same_v<...>
```
where the three first lines are fine but the last two get misinterpreted as doing relative includes.

**Link**
https://godbolt.org/z/8obx7YWYT

**Desktop**
 - OS: Win 10
 - Browser Firefox
 - Version Latest

**Additional context**

### PR DESCRIPTION
**Pull request by RubenRBS**
This seems to be working, but I might be missing some edge case somewhere

Closes #3045

### CODE REVIEW

File: lib/base-compiler.js [hunk 1]
```diff
@@ -1733,7 +1733,7 @@ Please select another pass or change filters.`;
     // to simple attempts with a clear diagnostic; the service still needs to
     // assume that malicious actors can make the compiler open arbitrary files.
     checkSource(source) {
-        const re = /^\s*#\s*i(nclude|mport)(_next)?\s+["<](\/|.*\.\.)[">]/;
+        const re = /^\s*#\s*i(nclude|mport)(_next)?\s+["<]((\.{1,2}|\/)[^">]*)[">]/;
```
Comment (RubenRBS) [inline lib/base-compiler.js:1736]: `./blah`
`../bleh`
`/blep`

should all be covered here. The `[^">]` part comes from https://github.com/compiler-explorer/compiler-explorer/blob/main/static/compiler-service.js#L270

**Review by supergrecko — APPROVED** [review]

**Comment by mattgodbolt** [comment]
I think this works: thanks!

File: test/base-compiler-tests.js [hunk 2]
```diff
@@ -84,6 +84,25 @@ describe('Basic compiler invariants', function () {
         should.exist(badSource);
         badSource.should.equal('<stdin>:1:1: no absolute or relative includes please');
     });
+    it('should not warn of path-likes outside C++ includes (Bug #3045)', () => {
```
Comment (mattgodbolt) [inline test/base-compiler-tests.js:87]: You're the best @RubenRBS - thanks!

**Review by mattgodbolt — APPROVED** [review]

**Comment by partouf** [comment]
this is now live

**Pull request merged**